Katello/katello
# Fixes #27942 - docker distributors have wrong base_path

This is needed to make mirror syncing work for Docker repositories that have custom naming enabled in the environment.

Note, this is a Pulp 3 bug.

To test:
1) Sync some Docker repo.
2) Make a lifecycle environment and set the "Registry Name Pattern" to something like `<%= organization.label %>-<%= repository.docker_upstream_name%>AaBbCc`.
3) Create a content view and add the Docker repo.  
4) Publish and promote the content view to your lifecycle environment.
5) Point a mirror/capsule/smart proxy with content to your lifecycle environment and sync it.

The smart proxy sync should happen without any issues.
